<launch>
    <arg name="bag_file" default="/media/eugeniu/T7/2024-03-07-14-40-23.bag"/>
    
    <!--  
    <node pkg="rosbag" type="play" name="rosbag_play" args=" clock -r .6 -s 0. $(arg bag_file) " output="log"/>
    -->

    <arg name="rviz" default="true" />
    <arg name="rviz_car" default="false" />

    <rosparam command="load" file="$(find s_mls)/config/hesai.yaml" />

    <param name="max_iteration" type="int" value="3" /> <!--3-->
    <param name="Bag" type="string" value="$(arg bag_file)" />
    
    <param name="filter_size_surf" type="double" value=".5" />
    <param name="filter_size_map" type="double" value=".5" />

    <param name="cube_side_length" type="double" value="120" /> <!--1000-->

    <node pkg="s_mls" type="mls_mapping" name="mls_mapping" output="screen" /> 
    
    <group if="$(arg rviz)">
        <group if="$(arg rviz_car)">
            <include file="$(find s_mls)/launch/robot_car.launch">
                <arg name="fixed_frame" value="MLS"/>
            </include>
        </group>
        <node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find s_mls)/rviz_cfg/visualize.rviz" /> 
    </group>

    
    <node pkg="hector_trajectory_server" type="hector_trajectory_server" name="trajectory_server" ns="MLS" >
        <param name="/target_frame_name" value="world" />
        <param name="/source_frame_name" value="MLS" />
        <param name="/trajectory_update_rate" value="15.0" />
        <param name="/trajectory_publish_rate" value="15.0" />
    </node>

    <node pkg="hector_trajectory_server" type="hector_trajectory_server" name="trajectory_server_GPSFix" ns="GPSFix" >
        <param name="/target_frame_name" value="world" />
        <param name="/source_frame_name" value="GPSFix" />
        <param name="/trajectory_update_rate" value="15.0" />
        <param name="/trajectory_publish_rate" value="15.0" />
    </node>

</launch>
